### PR TITLE
Jra am v3.0 psd01

### PR DESCRIPTION
**Pull request by jamsden**
Same changes as made for CM, RM and QM.

### CODE REVIEW

**Comment by ndjc** [comment]
Note that this review did uncover a defect in ShapeChecker, which I have corrected in a commit to the exiting branch sc-issue126. @berezovskyi, if you could review that PR #371, we can get the correct checks into master. Note that some specs will fail unless we either remove the -C flag from the check script for that spec, or fix the spec to have the right metadata on both vocals and shapes.

**Comment by berezovskyi** [comment]
For the record, I approved the PR in question and it was merged.

File: specs/am/architecture-management-shapes.ttl [hunk 1]
```diff
@@ -20,60 +20,72 @@
...
 # OSLC Architecutre Management Resource Shapes
 
-<#Resource>
+:
+  a oslc:ResourceShapeConstraints ;
+  dcterms:title "The OSLC Architecture Management(AM) Coonstraints" ;
+  rdfs:label "The OSLC Architecture Management(AM) Coonstraints" ;
+  dcterms:description "All vocabulary URIs defined in the OSLC Architecture Management (AM) namespace."^^rdf:XMLLiteral ;
+  dcterms:publisher <https://open-services.net/about/> ;
+  dcterms:issued "2020-xx-xx"^^<http://www.w3.org/2001/XMLSchema#date> ;
```
Comment (berezovskyi) [inline specs/am/architecture-management-shapes.ttl:33]: should be removed until publication

File: specs/am/architecture-management-shapes.ttl [hunk 2]
```diff
@@ -20,60 +20,72 @@
...
-@base <http://open-services.net/ns/core/shapes/3.0/architecture-management-shape.ttl> .
+@prefix : <http://open-services.net/ns/am/shapes/3.0#> . 
 
 # OSLC Architecutre Management Resource Shapes
 
-<#Resource>
+:
+  a oslc:ResourceShapeConstraints ;
+  dcterms:title "The OSLC Architecture Management(AM) Coonstraints" ;
+  rdfs:label "The OSLC Architecture Management(AM) Coonstraints" ;
```
Comment (berezovskyi) [inline specs/am/architecture-management-shapes.ttl:30]: May I suggest "OSLC Architecture Management (AM) Coonstraints"? Note the dropped article and an added space before the abbreviation.

File: specs/core/oslc-core.html [hunk 3]
```diff
@@ -594,18 +594,18 @@ <h2>Resource Constraints</h2>
...
-            <code>http://open-services.net/ns/cm/shapes/3.0</code> with members such as:
+            For example, for Change Management, there should be a container at:
+            <code>http://open-services.net/ns/cm/shapes</code> with members such as:
           </p>
           <pre>
-        http://open-services.net/ns/cm/shapes/3.0/changerequest
-        http://open-services.net/ns/cm/shapes/3.0/somenewshape
-        http://open-services.net/ns/cm/shapes/2.0/unchangedshape
+        http://open-services.net/ns/cm/shapes/3.0#TaskShape
+        http://open-services.net/ns/cm/shapes/3.1#WeaknessShape
```
Comment (berezovskyi) [inline specs/core/oslc-core.html:607]: What is version 3.1?

File: specs/core/oslc-core.html [hunk 4]
```diff
@@ -594,18 +594,18 @@ <h2>Resource Constraints</h2>
...
+        http://open-services.net/ns/cm/shapes/2.0#ChangeRequestShape
```
Comment (berezovskyi) [inline specs/core/oslc-core.html:608]: Why still 2.0?

File: specs/core/oslc-core.html [hunk 5]
```diff
@@ -594,18 +594,18 @@ <h2>Resource Constraints</h2>
               an LDPC for all the shapes for a spec version, at a URI defined by the following pattern:
             </p>
             <pre>
-      http://open-services.net/ns/[vocab short name]/shapes/[SPEC-version]
+      http://open-services.net/ns/[vocab-short-name]/shapes/[SPEC-version]
```
Comment (berezovskyi) [inline specs/core/oslc-core.html:597]: Should be "http://open-services.net/ns/[vocab-short-name]/shapes/[SPEC-version]#" if that's a prefix declaration.

File: specs/core/core-vocab.ttl [hunk 6]
```diff
@@ -520,9 +523,9 @@ oslc:postBody
   rdfs:isDefinedBy oslc: ;
   rdfs:label "postBody" ;
   rdfs:comment """The body of a POST request to return the next page if the response was to a POST request.
- Where a paged resource supports POST with an application/x-www-form-urlencoded body as an alternative to GET to
- avoid the request URI exceeding server limitations, the oslc:ResponseInfo in the response to the POST SHOULD
- contain this property so that a client knows what to POST to get the next page.""" .
+Where a paged resource supports POST with an application/x-www-form-urlencoded body as an alternative to GET to
```
Comment (berezovskyi) [inline specs/core/core-vocab.ttl:526]: I think the whole exercise is futile given that not only whitespace is preserved but also line termination: https://www.w3.org/TR/turtle/#h4_turtle-literals

Just keep it all on a single line or don't bother with all the spacing and expect the reader to clean it up by stripping whitespace from every line and then joining the lines with a space.

File: specs/core/core-vocab.ttl [hunk 7]
```diff
@@ -24,11 +24,14 @@ oslc:
   a owl:Ontology ;
   dcterms:title "The OSLC Core Vocabulary" ;
   dcterms:description "All vocabulary URIs defined in the OSLC Core namespace."^^rdf:XMLLiteral ;
-  dcterms:modified "2020-06-11"^^<http://www.w3.org/2001/XMLSchema#date> ;
+  rdfs:label "The OSLC Core Vocabulary" ;
+  vann:preferredNamespacePrefix "oslc" ;
   dcterms:publisher <https://open-services.net/about/> ;
+  dcterms:issued "2020-xx-xx"^^<http://www.w3.org/2001/XMLSchema#date> ;
```
Comment (berezovskyi) [inline specs/core/core-vocab.ttl:30]: This should be commented out until the publication.

File: specs/core/core-shapes.ttl [hunk 8]
```diff
@@ -446,6 +460,14 @@
                           oslc:valueType           oslc:Resource ;
                           dcterms:description      "For object properties, the URI of a shape resource that describes the object resource."^^rdf:XMLLiteral
                         ] ;
+        oslc:property   [ a                        oslc:Property ;
```
Comment (berezovskyi) [inline specs/core/core-shapes.ttl:463]: What is this? Could you please clearly state all major changes in the PR description?

File: specs/core/core-shapes.ttl [hunk 9]
```diff
@@ -21,6 +21,20 @@
...
 @prefix oslc:  <http://open-services.net/ns/core#> .
 
+
+:
+  a oslc:ResourceShapeConstraints ;
+  dcterms:title "The OSLC Core Constraints" ;
+  dcterms:description "Constraints on vocabulary terms defined in the OSLC Core namespace."^^rdf:XMLLiteral ;
+  rdfs:label "The OSLC Core Constraints" ;
+  dcterms:publisher <https://open-services.net/about/> ;
+  dcterms:issued "2020-xx-xx"^^<http://www.w3.org/2001/XMLSchema#date> ;
```
Comment (berezovskyi) [inline specs/core/core-shapes.ttl:31]: Should be commented out until publication.

File: specs/cm/change-mgt-vocab.ttl [hunk 10]
```diff
@@ -28,10 +28,15 @@
 oslc_cm:
   a owl:Ontology ;
   dcterms:title "The OSLC Change Management(CM) Vocabulary" ;
+  rdfs:label "The OSLC Change Management(CM) Vocabulary" ;
   dcterms:description "All vocabulary URIs defined in the OSLC Change Management (CM) namespace."^^rdf:XMLLiteral ;
-  dcterms:source <https://tools.oasis-open.org/version-control/svn/oslc-ccm/trunk/specs/change-mgt/change-mgt-vocab.ttl> ;
   vann:preferredNamespacePrefix "oslc_cm" ;
-  rdfs:label "Change Management(CM)" .
+  dcterms:publisher <https://open-services.net/about/> ;
+  dcterms:issued "2020-xx-xx"^^<http://www.w3.org/2001/XMLSchema#date> ;
```
Comment (berezovskyi) [inline specs/cm/change-mgt-vocab.ttl:35]: Should be commented out until publication.

File: specs/cm/change-mgt-vocab.ttl [hunk 11]
```diff
@@ -28,10 +28,15 @@
 oslc_cm:
   a owl:Ontology ;
   dcterms:title "The OSLC Change Management(CM) Vocabulary" ;
+  rdfs:label "The OSLC Change Management(CM) Vocabulary" ;
```
Comment (berezovskyi) [inline specs/cm/change-mgt-vocab.ttl:31]: I would suggest dropping the The article and adding a space before the opening bracket here as well.

File: specs/cm/change-mgt-shapes.ttl [hunk 12]
```diff
@@ -20,246 +20,260 @@
...
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix : <http://open-services.net/ns/cm/shapes/3.0#> .
+
+:
+  a oslc:ResourceShapeConstraints ;
+  dcterms:title "The OSLC Change Management(CM) Constraints" ;
+  rdfs:label "The OSLC Change Management(CM) Constraints" ;
+  dcterms:description "Constraints on vocabulary terms defined in the OSLC Change Management (CM) namespace."^^rdf:XMLLiteral ;
+  dcterms:publisher <https://open-services.net/about/> ;
+  dcterms:issued "2020-xx-xx"^^<http://www.w3.org/2001/XMLSchema#date> ;
```
Comment (berezovskyi) [inline specs/cm/change-mgt-shapes.ttl:32]: Should be commented out until publication.

File: specs/am/architecture-management-vocab.ttl [hunk 13]
```diff
@@ -19,18 +19,21 @@
...
-    vann:preferredNamespacePrefix  "oslc_am" ;
-    # This needs to refer to the published source of the vocabulary, in the OASIS document library
-    dcterms:source        <https://raw.githack.com/oslc-op/oslc-specs/master/specs/am/architecture-management-vocab.ttl> .
+  a owl:Ontology ;
+  dcterms:title "The OSLC Architecture Management(AM) Vocabulary" ;
+  rdfs:label "The OSLC Architecture Management(AM) Vocabulary" ;
+  dcterms:description "All vocabulary URIs defined in the OSLC Architecture Management (AM) namespace."^^rdf:XMLLiteral ;
+  vann:preferredNamespacePrefix "oslc_am" ;
+  dcterms:publisher <https://open-services.net/about/> ;
+  dcterms:issued "2020-xx-xx"^^<http://www.w3.org/2001/XMLSchema#date> ;
```
Comment (berezovskyi) [inline specs/am/architecture-management-vocab.ttl:32]: Should be commented out until publication

File: specs/am/architecture-management-vocab.html [hunk 14]
```diff
@@ -38,7 +38,7 @@
           },
         ],
 
-        //publishDate:  "2018-10-09",
+        publishDate: "2020-08-31",
```
Comment (berezovskyi) [inline specs/am/architecture-management-vocab.html:41]: That would be a challenge given that the vote has not even opened yet and I did not see the ZIP of the artifacts.

File: specs/core/oslc-core.html (hunk 4 above)
Comment (jamsden) [reply specs/core/oslc-core.html:608]: Just an example, not intended to mean anything.

File: specs/core/oslc-core.html (hunk 3 above)
Comment (jamsden) [reply specs/core/oslc-core.html:607]: Just an example, not intended to mean anything other than to indicate what some future version of CM might look lik.

File: specs/core/oslc-core.html (hunk 5 above)
Comment (jamsden) [reply specs/core/oslc-core.html:597]: Its not necessarily a prefix definition, its a URI pattern.

File: specs/am/architecture-management-shapes.ttl (hunk 2 above)
Comment (jamsden) [reply specs/am/architecture-management-shapes.ttl:30]: yes

File: specs/cm/change-mgt-vocab.ttl (hunk 10 above)
Comment (jamsden) [reply specs/cm/change-mgt-vocab.ttl:35]: done

File: specs/am/architecture-management-shapes.ttl (hunk 1 above)
Comment (jamsden) [reply specs/am/architecture-management-shapes.ttl:33]: Commented out.

File: specs/cm/change-mgt-vocab.ttl (hunk 11 above)
Comment (jamsden) [reply specs/cm/change-mgt-vocab.ttl:31]: Done

File: specs/am/architecture-management-vocab.html (hunk 14 above)
Comment (jamsden) [reply specs/am/architecture-management-vocab.html:41]: commented out, will set when vote completes and before creating Zip file.

File: specs/am/architecture-management-vocab.ttl (hunk 13 above)
Comment (jamsden) [reply specs/am/architecture-management-vocab.ttl:32]: done

File: specs/core/oslc-core.html (hunk 5 above)
Comment (berezovskyi) [reply specs/core/oslc-core.html:597]: I am just afraid someone will use that to form URIs without #

File: specs/am/architecture-management-vocab.html (hunk 14 above)
Comment (berezovskyi) [reply specs/am/architecture-management-vocab.html]: The right sequence should be: create a PR and uncomment, create ZIP, initiate a vote, merge when passed.

File: specs/core/oslc-core.html (hunk 5 above)
Comment (jamsden) [reply specs/core/oslc-core.html:597]: That is the URI of an LDPC, so it shouldn't end in #. The examples below all show fragments of that URI. Now I'm not sure if that's why we used / before - the shape resource URI is an LDPC, not a document. Maybe we need to leave it a document and remove the reference to LDPC.

**Comment by jamsden** [comment]
I think this is now ready to merge. @berezovskyi ?

**Comment by jamsden** [comment]
I couldn't figure out how to resolve Nick's requested change. I did the change, just can't find a way to resolve it in GitHub.

**Comment by berezovskyi** [comment]
You need to merge master into your branch to resolve Nick's request. LGTM from my side.

**Pull request merged**